protocols/kad/src/kbucket.rs

```diff
@@ -600,6 +606,7 @@ mod tests {
     #[test]
     fn rand_distance() {
         fn prop(ix: u8) -> bool {
+            let ix = ix.min(254); // avoid overflow in 2^ix
```

Comment:
The overflow is not in `2^ix` but in `2^(ix +1)`, right? How about enforcing the bound below, i.e. in case it is `2^(255+1) - 1` do `u256::max` instead?

Reply:
Sorry, I don't get it.

Reply:
Would the following do the trick @kpp?

``` diff
modified   protocols/kad/src/kbucket.rs
@@ -605,7 +605,10 @@ mod tests {
             let b = U256::from(2);
             let e = U256::from(ix);
             let lower = b.pow(e);
-            let upper = b.pow(e + U256::from(1)) - U256::from(1);
+            let upper = b
+                .checked_pow(e + U256::from(1))
+                .map(|x| x - U256::from(1))
+                .unwrap_or(U256::MAX);
             lower <= n && n <= upper
         }
         quickcheck(prop as fn(_) -> _);
```

Reply:
Now I see. Thank you, done.